src/main/java/com/box/sdk/BoxMetadataFilter.java

```diff
@@ -0,0 +1,146 @@
+package com.box.sdk;
+import java.util.Date;
+
+import com.eclipsesource.json.JsonObject;
+/**
+ * <p>BoxMetadataFilter is used to help organize the request for when making metadata filter request
+ * in conjuction with search. The translation will look something like this:
+ * [{"templateKey":"marketingCollateral", "scope":"enterprise", "filters":{"documentType": "datasheet"}}]</p>
+ *
+ *
+ */
+public class BoxMetadataFilter {
+    private String templateKey;
+    private String scope = "enterprise";
+    private JsonObject filtersList;
+
+    /**
+     * Constructor for BoxMetadataFilter that initizlizes the JSON Object.
+     */
+    public BoxMetadataFilter() {
+        this.filtersList = new JsonObject();
+    }
+    /**
+     * Returns the template key that currently set.
+     * @return this.String template key.
+     */
+    public String getTemplateKey() {
+        return this.templateKey;
+    }
+    /**
+     * Set the current template key for the search filter.
+     * @param templateKey must be a metadata template key.
+     */
+    public void setTemplateKey(String templateKey) {
+        this.templateKey = templateKey;
+    }
+    /**
+     * return this.a list of the current filters that are being set.
+     * @return this.JsonObject filterList.
+     */
+    public JsonObject getFiltersList() {
+        return this.filtersList;
+    }
+    /**
+     * Set a filter to the filterList, example: key=documentType, value=special.
+     * @param key the key that the filter should be looking for.
+     * @param value the specific value that corresponds to the key.
+     */
+    public void setFilter(String key, String value) {
+        this.filtersList = new JsonObject();
```

Comment:
Can this class can only have one filter type set at a time? It looks like `filtersList` is being recreated in each setter.